### PR TITLE
Reimplement back links

### PR DESCRIPTION
**Pull request by nigel-lowry**
https://dfedigital.atlassian.net/browse/CAPT-200

Reimplements back links without recourse to JavaScript so back links now work in the Apple Safari browser

Removes breaking change created by https://github.com/DFE-Digital/claim-additional-payments-for-teaching/pull/1915

This stops another problem where a page is re-rendered as part of the wizard and allows a whole step to be skipped over to change previous answers (this isn't always possible with the JavaScript links in Chrome)

### CODE REVIEW

File: app/models/page_sequence.rb [hunk 1]
```diff
@@ -25,6 +25,17 @@ def next_slug
     slugs[current_slug_index + 1]
   end
 
+  def previous_slug
+    slug_index = current_slug_index
+    deadend_slugs = %w[complete existing-session eligible-now eligibility-confirmed eligible-later ineligible]
+
+    if slug_index.zero? || current_slug.in?(deadend_slugs)
```
Comment (vassyz) [inline app/models/page_sequence.rb:32]: It's a personal preference but I usually use a guard clause on a conditional that ends a method.

```ruby
return nil if slug_index.zero? || current_slug.in?(deadend_slugs)

slugs[slug_index - 1]
```

**Review by vassyz — APPROVED** [review]

File: app/models/page_sequence.rb (hunk 1 above)
Comment (kenfodder) [reply app/models/page_sequence.rb:32]: Yeah I prefer that as well, short circuit first

Comment (nigel-lowry) [reply app/models/page_sequence.rb]: Done, but it's one of the things I like about Ruby: you don't need to explicitly `return`

Comment (kenfodder) [reply app/models/page_sequence.rb]: Just a style, I'm not too fussed just convention and rubocop does pick these up on certain scenarios

**Review by kenfodder — APPROVED** [review]

**Review by kenfodder — APPROVED** [review]

**Pull request merged**